### PR TITLE
Add ADR 29: Simple Dropdown Task

### PR DESCRIPTION
**Pull request by srallen**
_Please request review from `@zooniverse/frontend` team. If PR is related to design, please request review from `@beckyrother` in addition._ 

Package: lib-classifier

Toward #1679

Affects #1742 and #1750.

Describe your changes:
Adds the draft ADR @snblickhan and @mrniaboc began and that I added some details to about the simple dropdown task that is being built and the changes from PFE we'd like to make. 

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by srallen** [comment]
I think we should add sample annotation JSON to the ADR.

**Comment by shaunanoordin** [comment]
Hello! I'm following up from my notes on PR https://github.com/zooniverse/front-end-monorepo/pull/1742#issuecomment-679435831

The TL;DR on the dev side is:

- There are two ways we can build a "simple dropdown" task.
  - **"-v2-"** we build a "simple" implementation on top of the classic `dropdown` task  
    - Pro: [already](https://github.com/zooniverse/front-end-monorepo/pull/1742), [built](https://github.com/zooniverse/front-end-monorepo/pull/1750).
    - Pro: can be improved further for backward compatibility. (The eventual return of cascading, yooo)
    - Con: we're still supporting old, complicated, and frankly unused data models that we want to phase out.
  - **"-v3-"** we build a new `dropdown-simple` task
    - Pro: will take only a few hours to build on monorepo. (I'm basically snipping code from v2)
    - Pro: forward-looking and more in line with the monorepo's goal of "building better things from the lessons from PFE"
    - Con: need to tinker with PFE's Project Builder.

@mrniaboc, @snblickhan - this isn't an exclusive choice btw. We can support both v2 and v3 plans, if we want to have our cake and eat it too. I feel the matter of dropdown implementation is a long-term software planning matter which, while _very important,_ shouldn't knock the Engaging Crowds project too far off track. Gotta balance that short-term stuff with the long-term thingamabobs. 🤷‍♂️ 

I think in dev terms, the real question is _"what's the priority of backwards compatibility?"_ which I think may be its own kettle of fish. (Tagging @srallen, @camallen)

(What happened to v1, you say? Don't ask, that was an over-ambitious prototype that went sideways.)

**Comment by shaunanoordin** [comment]
@srallen I've got you covered for data models!

### Current Data Models

Current `dropdown` task Task example (for a simple dropdown):

```
"T0":{
  "help":"",
  "next":"T1",
  "type":"dropdown",
  "selects":[
    {
      "id":"070b610fbf5d9",
      "title":"Favourite colour",
      "options":{
        "*":[
          {
            "label":"Red",
            "value":"hashed-value-R"
          },
          ...
        ]
      },
      "required":false,
      "allowCreate":false
    }
  ],
  "instruction":"Choose your favourite things"
}
```

Current `dropdown` task Annotation example:

```
"annotations":[
  {
    "task":"T0",
    "value":[
      {
        "value":"hashed-value-R",
        "option":true
      }
    ]
  }
]
```

### Proposed `dropdown-simple` Data Models

Based on your comments in https://github.com/zooniverse/front-end-monorepo/pull/1742#pullrequestreview-473755064

Proposed `dropdown-simple` task Task example:

```
"T0":{
  "help":"",
  "next":"T1",
  "type":"dropdown-simple",
  "options":[
    "Red",
    "Blue",
    ...
  ],
  "required":false,
  "allowCreate":false,
  "instruction":"Choose your favourite colour"
}
```

- ⚠️  In the above example, `selects[n].title` becomes redundant. Note the resulting change in the UI.
- Sarah: _[Dropping the hash value from the task model and from the annotation model. The purpose of the hash value is to support cascading, which the simple dropdown task will not do.](https://github.com/zooniverse/front-end-monorepo/pull/1742#pullrequestreview-473755064)_ + _[The options should become an array](https://github.com/zooniverse/front-end-monorepo/pull/1742#pullrequestreview-473755064)_
  - Shaun: makes sense for a simplified model. Just an array of strings works.
- Sarah: _[allowCreate is tentatively optional but will default to false](https://github.com/zooniverse/front-end-monorepo/pull/1742#pullrequestreview-473755064)_
  - Shaun: sounds legit.

Proposed `dropdown-simple` task Annotation example:

```
"annotations":[
  {
    "task":"T0",
    "value": {
      "value": 1,  // Corresponds to "Blue"
      "option":true
    }
  },
]
```

- Sarah: _[The annotation value object includes a key set to index of the option picked similar to the single choice task. Then another key still recording if this is a free text option. This is so we can support translations of the options.](https://github.com/zooniverse/front-end-monorepo/pull/1742#pullrequestreview-473755064)_
  - Shaun: cool.
- Sarah: _[If in the future, we support free text entry, then the annotation value could be: { value: { selection: 'free text entry', option: false } }](https://github.com/zooniverse/front-end-monorepo/pull/1742#pullrequestreview-473755064)_
  - Shaun: ice cool.

This is basically a kinda like a SingleChoiceTask, except the `.next` task-forking logic is replaced with a maybe-possibly free text option. (Wait, how do you code in MST, `types.model('SimpleDropdownAnnotation', { value: types.maybeNull(types.number OR types.string) })` ? )

File: docs/arch/adr-29.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+``` json
+
+```
+
+## Decision
+ 
+We will develop a simplified dropdown task in the new classifier in the FEM. Creating a simple dropdown task contains the following functionality:
+
+- Limited dropdown list options of a minimum (4) and of a maximum (20) number of options (justification: if less than 4, this can be a single choice task using radio buttons)
+- No cascading, dependencies, or affect on other select inputs
```
Comment (eatyourgreens) [inline docs/arch/adr-29.md:31]:
```suggestion
- No cascading, dependencies, or effect on other select inputs
```

**Comment by eatyourgreens** [comment]
#1742 has discussion about the migration pathway for legacy projects, like Notes from Nature, starting here:
https://github.com/zooniverse/front-end-monorepo/pull/1742#issuecomment-679903033

I think it would be useful to have some notes about the migration pathway in this ADR, even if it's just ideas for how we might port projects across.

**Comment by srallen** [comment]
This is still on my back burner to update and finish. Gonna have a look maybe next week.

**Comment by srallen** [comment]
@eatyourgreens and @shaunanoordin I finally tidied this up and is ready for review.

**Comment by srallen** [comment]
This is now ADR 30. Updated in the files.

**Comment by eatyourgreens** [comment]
Looks good to me. Any objections to my adding an example translation for a dropdown task? Every Name Counts is using dropdown menus in German and English.

File: docs/arch/adr-30.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+- Dates will need to be set up manually, e.g. free-text for Day, researcher-generated list of months for Month. Note that for Engaging Crowds project #1, each subject set will cover the same year, so that info should already be in the subject metadata. In the future, we will develop some kind of date input task, but it won’t be available for the initial Engaging Crowds effort.
+- Generally project owners using the PFE dropdown task will need to migrate to one of the new options, and thus there will not be backwards compatibility with the PFE version.
+- The updated task type will need an updated editor in the lab.
```
Comment (eatyourgreens) [inline docs/arch/adr-30.md:133]: I think the new task type will also need translatable strings defined in Panoptes, so that the task question and options are available in the workflow translations editor.

Comment (eatyourgreens) [reply docs/arch/adr-30.md:133]: Here's an example of the translation keys for a single dropdown with the PFE task. I think the new task will simplify the keys.
https://pandora.zooniverse.org/#/project/908/workflow/3339?language=es

Spanish translations for a single dropdown menu, showing the key names for the task help, instruction and option labels.

**Comment by srallen** [comment]
@eatyourgreens feel free to add information about translations in the consequences section.

**Comment by eatyourgreens** [comment]
I've added language strings to the examples of a PFE dropdown, and a note that the new task might need updates to the translatable fields in Panoptes.

**Review by eatyourgreens — APPROVED** [review]
